### PR TITLE
Add DPI Awareness configuration options

### PR DESCRIPTION
**Pull request by MaartenBent**
Add options to CMake and configure to set the DPI awareness on Windows. 

The default DPI awareness is set to `system`.

I could not find a way to set the DPI Awareness in the generated VS project files and `makefile.vc`. These rely on a default manifest and do not use the manifest defined in the resource file.
Modifying a project file manually and adding `AdditionalManifestFiles` to `VCManifestTool` does not seem to work, and I don't know if bakefile supports anything like this.
`makefile.vc` would probably need something with `mt.exe` as described in https://github.com/wxWidgets/wxWidgets/pull/1602#issuecomment-546341065. Does bakefile support this?

I added it to the `common_settings` template in the bakefile, I should probably in `wx_util` template but I couldn't get it to bake successfully. I did not commit the rebaked files, because it reverts using response files.

I also fixed a warning I encountered with testing `makefile.gcc`.

### CODE REVIEW

File: configure.in [hunk 1]
```diff
@@ -1124,6 +1126,14 @@ if test -n "$wxWITH_CXX"; then
...
 fi
 
+USE_DPI_AWARE_MANIFEST=0
+if test "$wxWITH_DPI_MANIFEST" = "system" ; then
+    USE_DPI_AWARE_MANIFEST=1
+fi
+if test "$wxWITH_DPI_MANIFEST" = "per-monitor" ; then
+    USE_DPI_AWARE_MANIFEST=2
+fi
+
```
Comment (vadz) [inline configure.in]: I'd replace this with
```sh
case "$wxWITH_DPI_MANIFEST" in
    system) USE_DPI_AWARE_MANIFEST=1 ;;
    per-monitor) USE_DPI_AWARE_MANIFEST=2 ;;
    *) AC_MSG_WARN([Unsupported DPI awareness value "$wxWITH_DPI_MANIFEST" ignored.])
esac
```

**Review by vadz — COMMENTED** [review]
Thanks a lot for implementing this!

But I wonder if, even though this option doesn't affect the library compilation, `wxUSE_DPI_AWARE_MANIFEST` still shouldn't be defined in `wx/msw/setup_inc.h` and documented in `docs/doxygen/mainpages/const_wxusedef.h` together with all the rest of them? Maybe not the former (although it could be useful to change the default for all apps using the given wx build perhaps?), but I think the latter needs to be done because we need to document it somewhere and I don't see a better place.

What do you think?

**Comment by MaartenBent** [comment]
Adding it to `wx/msw/setup_inc.h` (and generated `setup.h`) is useless, because `wx.rc` does not include `setup.h`. It has to be defined via a command-line argument to `WINDRES`.

The default behavior can be changed by modifying/overriding `USE_DPI_AWARE_MANIFEST` in `build/msw/config.[bcc,gcc]`.

I see bakefile has a [res-define](https://bakefile.org/doc_legacy/ch04.html#tag.res-define), I'll use this instead of `define`.

**Comment by RadMap** [comment]
With risk of showing my ignorance in this matter: Since cmake 3.16 (I believe) we have access to the VS_DPI_AWARE property in cmake that might come in handy. Just thought it might be worth mentioning.

**Comment by vadz** [comment]
That's great, thanks Maarten! I'll merge this, if there is anything to be done related to `VS_DPI_AWARE`, it could be done in a separate PR later.

**Comment by vadz** [comment]
> We only support Per-Monitor V2. This can't be set in the Visual Studio project properties or via `VS_DPI_AWARE`, only via the manifest.

That was my impression as well, but I find it utterly weird that even MSVS 2019 doesn't support per-monitor v2 for its projects. I wonder what do the developers of native desktop applications actually do (I assume there must still be some of them left...)?

**Pull request merged**